flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -616,6 +622,11 @@ public void sendTaskEvent(TaskEvent event) throws IOException {
 		}
 	}
 
+	@Override
+	public void unblockCheckpoint(int channelIndex) {
+		channelIndexToInputChannel[channelIndex].unblockCheckpoint();
```

Comment:
`channelIndexToInputChannel` should be protected in sync by `requestLock`, because in code path this array would be modified by RPC thread via `updateInputChannel`. Although that method is never invoked in streaming job, it still has the potential risk to break this assumption.